includes/class-give-email-access.php

```diff
@@ -255,6 +261,8 @@ public function check_for_token() {
 				}
 			}
 
+			// Set Receipt Access Session.
+			Give()->session->set( 'receipt_access', true );
```

Comment:
@mehul0810 Reciept access depends upon `payment_key`, right? This should be set conditionally, right?

Reply:
@ravinderk Yes.. It is dependent on it and it is set conditionally only. See the condition on L256 `if( ! empty($token)){`